packages/react-core/src/components/Progress/ProgressHelperText.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import progressStyle from '@patternfly/react-styles/css/components/Progress/progress';
+import { HelperText, HelperTextItem, HelperTextItemProps } from '../HelperText';
+import { ProgressProps } from './Progress';
+
+export interface ProgressHelperTextProps extends React.HTMLProps<HTMLDivElement> {
+  /** Content which can be used to convey additional information about the progress component.
+   * Renders content inside of our helper text component with automatic helper text variant setting based on the
+   * variant of the progress component when a string is passed, otherwise renders the content directly.
+   */
+  children?: React.ReactNode;
+  /** Variant of the progress component. Only applies if children is a string. */
+  progressVariant?: ProgressProps['variant'];
+}
+
+export const ProgressHelperText: React.FunctionComponent<ProgressHelperTextProps> = ({
+  children,
+  progressVariant,
+  ...props
+}: ProgressHelperTextProps) => {
+  const helperTextVariantMapping: {
+    [Property in ProgressProps['variant']]: HelperTextItemProps['variant'];
+  } = {
+    success: 'success',
+    warning: 'warning',
+    danger: 'error'
+  };
+
+  const helperTextIsBasic = typeof children === 'string';
+
+  const basicHelperText = (
+    <HelperText>
+      <HelperTextItem variant={helperTextVariantMapping[progressVariant]}>{children}</HelperTextItem>
+    </HelperText>
+  );
+
+  return (
+    <div className={progressStyle.progressHelperText} {...props}>
+      {helperTextIsBasic ? basicHelperText : children}
```

Comment:
Why do we not render this in a `HelperText` component? Is it so that the consumer can pass in a HelperText component themselves?

Reply:
Correct, with this approach consumers can pass a `HelperText` (or multiple `HelperText`s) with custom options like `isLiveRegion` or a custom `variant`, or potentially different components rather than a `HelperText`.

Reply:
@mcarrano What are you thought on having the "helperText be flexible? They way we have it implemented here, the consumer can pretty much have the helper the be anything.  If we are ok with the flexibility, I would suggest we not even have the component internally `HelperText` internally based on the string and just let the consumer pass it in.

Reply:
I think I'm OK with the consumer passing this in as it allows for use cases we have not considered. We do have a Helper Text component: https://www.patternfly.org/v4/components/helper-text Does this leverage that? Is the concern that this does not cover all relevant use cases?

Reply:
@mcarrano yes the current implementation allows consumers to pass a simple string and will automatically pass that string to a `HelperText` component and control the `HelperText` variant based on the variant of the `Progress`.

But it also allows the consumer to pass in a `HelperText` with a different configuration, or any other component they would like, if they have a use case for which the more automatic approach mentioned above doesn't work

Reply:
That sounds fine to me. @tlabaj did you have concerns?

Reply:
I actually think it is a little unexpected that only a string would cause the component to create the `HelperText` since the `HelperTextItem` type is a react node.